Rpi cn0508

mirceacaprioru opened the pull request:
For starters I would like just a general feedback for this PR. No need to merge this set just yet. 

Some commits refer only to the CN0508 application, while the last 3 commits modify the ad7124 driver and introduce a YAML dt-binding documentation. This part is intended to be upstreamed.

mirceacaprioru commented:
Changelog v1 -> v2

- added patternProperties for channel subnodes in YAML format.

mirceacaprioru commented:
Changelog v2 -> v3
- add adi,buffered-negative and adi,buffered-positive properties

commodo commented on arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts:
```diff
@@ -109,21 +109,21 @@
-		#size-cells = <0>;
-		spi-max-frequency = <125000000>;
-	};
+//	spidev0: spidev@0{
```
curios: why is this done?
it seems to change some upstream code

commodo replied:
FWIW: you can try to disable spidev devices in overlays with
```
&spidev0 { 
     status = "disabled";
}
```

[ if this is what you wanted to do]

mirceacaprioru replied:
This is just something particular to the cn0508 project. won't upstream this part feel free to ignore. It is done because the CS are also used by the spidev and it gives a warning when you use them again for ad7124 and ad5683.

mirceacaprioru replied:
ack, thanks for the this that one slipped my attention.

commodo commented on drivers/iio/adc/ad7124.c:
```diff
@@ -463,13 +463,6 @@ static int ad7124_of_parse_channel_config(struct iio_dev *indio_dev,
 		if (ret)
 			goto err;
 
-		if (ain[0] >= st->chip_info->num_inputs ||
```
maybe update the title to `iio: adc: ad7124: Remove input number limitation`
it's more inline with what the patch does

commodo commented on drivers/iio/adc/ad7124.c:
```diff
@@ -526,8 +534,13 @@ static int ad7124_setup(struct ad7124_state *st)
 
+		tmp = (st->channel_config[i].buf_positive << 1)  +
+			st->channel_config[i].buf_negative;
+		pr_err("ad7124 buffer = %d", tmp);
```
this looks like a forgotten debug statement

commodo commented on drivers/iio/adc/ad7124.c:
```diff
@@ -482,6 +489,7 @@ static int ad7124_of_parse_channel_config(struct iio_dev *indio_dev,
 			of_property_read_bool(child, "adi,buffered-positive");
 		st->channel_config[channel].buf_negative =
 			of_property_read_bool(child, "adi,buffered-negative");
+
```
nitpick: patch adds extra line; maybe move this into the previous patch that adds buffered support

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7124.yaml:
```diff
@@ -0,0 +1,144 @@
+patternProperties:
+  "^channel@[01]$":
+    type: object
+    description: Represents the external channels which are connected to the ADC.
```
maybe reference `Documentation/devicetree/bindings/iio/adc/adc.txt` here as well

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7124.yaml:
```diff
@@ -0,0 +1,144 @@
+
+      channel@3 {
+        reg = <3>;
+         diff-channels = <6 7>;
```
these 2 look misaligned

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7124.yaml:
```diff
@@ -0,0 +1,144 @@
+examples:
+  - |
+    adc@0 {
+      compatible = "adi,ad7124-4";
```
generally: this yaml file, won't likely build;
maybe try it on a newer tree?

mirceacaprioru commented:
Changelog v3 -> v4
- removed commit that comments out spidev 
- added status disabled for spidev in the cn0508 overlay
- referenced documantation in the YAML implementation
- removed debug print
